File: cpp/57a_1220A_Cards.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n, no_z = 0, no_n = 0;
    std::string s, ans = "";
    
    cin>>n>>s;
    
    for(auto i : s){
        if (i == 'z')
            no_z++;
        else if(i == 'n')
            no_n++;
    }

    while(no_n--){
        ans += "1 ";
    }
    while(no_z--){
        ans += "0 ";
    }
    
    cout<<ans;

    return 0;
}

File: python/57a_1220A_Cards.py
def solve(n, z):
    ans = ""
    while n:
        n -= 1
        ans += "1 "
    while z:
        z -= 1
        ans += "0 "
    return ans

def main():
    n = int(input())
    s = input()

    no_z, no_n = 0, 0
    for i in s:
        if i == "n":
            no_n += 1
        elif i == "z":
            no_z += 1
    
    ans = "" + solve(no_n, no_z)
    print(ans)

if __name__ == "__main__":
    main()